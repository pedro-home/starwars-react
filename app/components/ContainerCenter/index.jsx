import React from 'react';
import PropTypes from 'prop-types';

const Component = ({ children }) => {
  return <div className="container-center">{children}</div>;
};

Component.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
};

export default Component;
