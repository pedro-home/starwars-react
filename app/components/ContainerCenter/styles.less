.container-center {
  top: 50%;
  position: relative;
  transform: translateY(-50%);

  @media (min-width: @screen-phone) {
    left: 50%;
    position: absolute;
    transform: translate(-50%, -50%);
  }
}
