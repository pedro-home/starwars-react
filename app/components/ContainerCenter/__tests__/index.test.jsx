import React from 'react';
import { render } from '@testing-library/react';

import ContainerCenter from '../index';

describe('Given <ContainerCenter />', () => {
  let component;
  let defaultProps;
  let text;

  describe('When initializes', () => {
    beforeEach(() => {
      text = 'Hello';
      defaultProps = { children: <span>{text}</span> };
      const { container } = render(<ContainerCenter {...defaultProps} />);
      component = container;
    });

    it('Then should match the snapshot', () => {
      expect(component).toMatchSnapshot();
    });

    it('And should render children', () => {
      expect(component.querySelector('span').innerHTML).toEqual(text);
    });
  });
});
