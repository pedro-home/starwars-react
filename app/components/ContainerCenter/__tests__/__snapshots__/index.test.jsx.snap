// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <ContainerCenter /> When initializes Then should match the snapshot 1`] = `
<div>
  <div
    class="container-center"
  >
    <span>
      Hello
    </span>
  </div>
</div>
`;
